File: test/cpp-tableau-loader/src/hub/custom/item/custom_item_conf.cpp
#include "hub/custom/item/custom_item_conf.h"

#include "logger.pc.h"

const std::string CustomItemConf::kCustomName = "CustomItemConf";

bool CustomItemConf::ProcessAfterLoadAll(const tableau::Hub& hub) {
  auto conf = hub.Get<protoconf::ItemConfMgr, protoconf::ItemConf::Item>(1);
  if (!conf) {
    ATOM_ERROR("hub get item 1 failed!");
    return false;
  }
  special_item_conf_ = *conf;  // value copy
  ATOM_DEBUG("custom item conf processed");
  return true;
}

const std::string& CustomItemConf::GetSpecialItemName() const { return special_item_conf_.name(); }

File: test/go-tableau-loader/customconf/custom_item_conf.go
package customconf

import (
	"fmt"

	"github.com/tableauio/loader/test/go-tableau-loader/protoconf"
	tableau "github.com/tableauio/loader/test/go-tableau-loader/protoconf/loader"
)

const CustomItemConfName = "CustomItemConf"

type CustomItemConf struct {
	tableau.UnimplementedMessager
	specialItemConf *protoconf.ItemConf_Item
}

func (x *CustomItemConf) Name() string {
	return CustomItemConfName
}

func (x *CustomItemConf) ProcessAfterLoadAll(hub tableau.MessagerContainer) error {
	config, err := hub.GetItemConf().Get1(1)
	if err != nil {
		return err
	}
	x.specialItemConf = config
	fmt.Println("custom item conf processed")
	return nil
}

func (x *CustomItemConf) GetSpecialItemName() string {
	return x.specialItemConf.GetName()
}

func init() {
	tableau.Register(func() tableau.Messager {
		return new(CustomItemConf)
	})
}
